game-core/src/main/java/games/strategy/engine/framework/ui/GameChooserModel.java

```diff
@@ -51,66 +53,55 @@ public GameChooserEntry get(final int i) {
     return super.get(i);
   }
 
-  static Set<GameChooserEntry> parseMapFiles() {
-    final Set<GameChooserEntry> parsedMapSet = new HashSet<>();
-    FileUtils.listFiles(ClientFileSystemHelper.getUserMapsFolder())
+  public static Set<GameChooserEntry> parseMapFiles() {
+    return FileUtils.listFiles(ClientFileSystemHelper.getUserMapsFolder())
         .parallelStream()
-        .forEach(
```

Comment:
This could easily been a race-condition, there's no guarantee that parsedMapSet can handle multiple Threads adding elements